src/app.md

```diff
@@ -725,6 +726,80 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-semver-latest]: #ex-semver-latest
+<a name="ex-semver-latest"></a>
+## Find the latest version satisfying given range
+[![semver-badge]][semver] [![cat-config-badge]][cat-config]
+
+Given a list of version &strs, finds the latest [`semver::Version`] that satisfying a given [`semver::VersionReq`] using [`VersionReq::matches`].
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate semver;
+
+use semver::{AlphaNumeric, Numeric, Version, VersionReq};
+use std::vec::IntoIter;
+#
+# error_chain! {
+#     foreign_links {
+#         SemVer(semver::ReqParseError);
+#     }
+# }
+
+fn find_max_matching_version(
+    version_req_str: &str,
+    version_str_intoiter: IntoIter<&str>,
+) -> Result<Option<Version>> {
+    let vreq = VersionReq::parse(version_req_str)?;
+
+    Ok(
+        version_str_intoiter
+        //.iter()
+        .filter_map(|s| Version::parse(s).ok())
+        .filter(|s| vreq.matches(s))
+        .max(),
+    )
+}
+
+fn run() -> Result<()> {
+    assert_eq!(
+        find_max_matching_version("<= 1.0.0", vec!["0.9.0", "1.0.0", "1.0.1"].into_iter())?,
+        Some(Version {
+            major: 1,
+            minor: 0,
+            patch: 0,
+            pre: vec![],
+            build: vec![],
+        })
+    );
+
+    assert_eq!(
+        find_max_matching_version(
+            ">1.2.3-alpha.3",
+            vec![
+                "1.2.3-alpha.3",
+                "1.2.3-alpha.4",
+                "1.2.3-alpha.10",
+                "1.2.3-beta.4",
+                "3.4.5-alpha.9",
+            ].into_iter()
+        )?,
+        Some(Version {
```

Comment:
Since the expected result can be surprising, I wonder if a comment should be added like: "Shows Semver precedence for pre-release tags"